lib/shared/inference/openai_image_generator.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module ::DiscourseAi
+  module Inference
+    class OpenAiImageGenerator
+      TIMEOUT = 60
+
+      def self.perform!(prompt, model: "dall-e-3", size: "1024x1024", api_key: nil)
+        api_key ||= SiteSetting.ai_openai_api_key
+
+        url = URI("https://api.openai.com/v1/images/generations")
+        headers = { "Content-Type" => "application/json" }
+        headers["Authorization"] = "Bearer #{api_key}"
+
+        payload = { model: model, prompt: prompt, n: 1, size: size, response_format: "b64_json" }
+
+        Net::HTTP.start(
```

Comment:
Should we maybe use our `FinalDestination::HTTP` library here instead, since it wraps `Net::HTTP`?

https://github.com/discourse/discourse/blob/d6b9c8c828eb48b0e4bc237563fb10aab1ad211b/lib/final_destination.rb#L130-L146

Reply:
It automatically follows redirects etc

Reply:
no need to follow redirects here, I kind of prefer to keep this code trivial and use net http

Reply:
I mostly mean that, nowhere in core do we use `Net::HTTP` directly, it's always `FinalDestination::HTTP`, so it would be good to be consistent.